### ISSUE TITLE
[Bug]: FTBQuest Not completing/Crashing when using "Observation" type

### ISSUE THREAD

**Issue by alfredggttv** [FTBTeam/FTB-Mods-Issues#629]
### Mod

FTB Quests

### Mod version

ftb-quests-forge-1902.4.5-build.156

### Forge / Fabric version

1.19.2 43.2.3

### Modpack & version

ATM8 1.0.8

### What issue are you having?

I'm trying to create a quest for observing a certain type of entity with a type tag, notably the custom Productive Bees that are added for the pack (Allthemodium, etc). 

I've set the type to `entity_type_tag` and the Observe to `productivebees:allthemodium` and upon observing the Bee, the quest does not complete.

Adding any kind of #, ', or {} characters into the `Observe` field causes the entire game to crash when looking at the Bee.

While I could 100% be using the wrong format for an entity_type_tag, having the game crash while observing a Bee should not happen. The suggestion would be to not let the field complete when inputting the `entity_type_tag` with improper format. 

### Crashlogs

Latest Crash Log: 
https://mclo.gs/UZEnIrn
https://pste.ch/isugagewiz.pl
[crash-2023-01-14_11.39.50-client.txt](https://github.com/FTBTeam/FTB-Mods-Issues/files/10418466/crash-2023-01-14_11.39.50-client.txt)

### Steps to reproduce

1. Create quest using `Observation` type requirement.
2. Change type to `entity_type_tag`
3. Input Entity Type Tag. In my case, `productivebees:allthemodium`.
4. Observe Allthemodium Bee.
5. Quest does not complete.

For the crash:

Follow steps 1 and 2 above.
3. Input Entity Type Tag with a #, #productivebees:allthemodium.
4. Observe Bee.
5. Crash.

**Comment by desht** [comment]
You're right, it shouldn't crash in this situation. The correct syntax is simply `productivebees:allthemodium`, without the leading "#".

Next release will tolerate (and ignore) the leading "#" if present, and additionally not crash if a bad resource location is supplied.

As far as the current tag not being recognised by the task: looking at the code, this should work, assuming the bee you're looking at is definitely in the `productivebees:allthemodium` entity tag.  Where does this tag come from, i.e. who defines it, and what's in it?  Doesn't appear to be defined by the Productive Bees mod itself...

**Comment by desht** [comment]
As a test, I created an observation task for the entity type tag `minecraft:frog_food`, which includes slimes and magma cubes).  The task correctly completed upon observing a slime (but not other entities), so I'd say the code is working correctly.  I would suggest checking again that the `productivebees:allthemodium` tag really includes the bee you want to observer.

**Comment by alfredggttv** [comment]
Will double check on the tag to make sure it's referencing the correct bee. Productive bees add customizable bees with the tag: productivebees:configurable_bee. The "Allthemodium" part is the NBT tag that is attached to it. That tag for the Allthemodium Bee is productivebees:allthemodium.

If I were to look at the Allthemodium Bee, it just shows that I'm looking at "Productivebees:configurable_bee".

Using the Strong NBT Filter, this is what is given to me when I check the spawn egg for it. 
![image](https://user-images.githubusercontent.com/107154056/212984737-3716cc7d-edb2-4ef5-9db6-bfee009e0bfb.png)

**Comment by alfredggttv** [comment]
Upon further research, setting the type to entity_type, then setting the "Observe" to `productivebees:configurable_bee` does allow me to complete the quest, but that would work for every kind of added bee.

When I switch to entity_type_tag, I've tested both `productivebees:allthemodium` and `productivebees:allthemodium_bee` and neither completes the quest. 

Using `/data get entity` for it, it comes up with this:
![image](https://user-images.githubusercontent.com/107154056/212987589-7963132a-6f9c-46e7-891d-191fdaed9ec5.png)

**Comment by desht** [comment]
That's the entity type, not the entity type tag.  So using an observation type of "Entity Type" and a name of "productivebees:allthemodium" should work here.

**Comment by alfredggttv** [comment]
![image](https://user-images.githubusercontent.com/107154056/213076792-9263b0cb-06a4-406c-a290-008bbde9a7c3.png)
Here is what the quest is set to.

![2023-01-17_21 29 57](https://user-images.githubusercontent.com/107154056/213076882-f19292e7-2809-4781-9bd5-7ed98a32e669.png)

Here is me observing the bee, and it not completing. I have also tried on the other types. The bee does complete when `Entity_Type` is set to `productivebees:configurable_bee`.

**Comment by desht** [comment]
Ah, I see - `productivebees:allthemodium` is just an NBT tag on the entity, whose entity type is `productivebees:configurable_bee`.

Unfortunately the Observation task doesn't support inspection of entity NBT at this time, although that's something that could be added in the next release...
